src/main/java/uk/ac/ebi/embl/converter/fftogff3/GFF3DirectivesFactory.java

```diff
@@ -42,37 +37,37 @@ private String buildTaxonomyUrl(Optional<OrganismQualifier> qualifier) {
                 .orElseGet(getOrganism);
     }
 
-    public GFF3Directives.GFF3Species extractSpecies(Entry entry) throws NoSourcePresentException {
+    public GFF3Species extractSpecies(Entry entry) throws NoSourcePresentException {
 
         Feature feature =
                 Optional.ofNullable(entry.getPrimarySourceFeature()).orElseThrow(NoSourcePresentException::new);
 
         Optional<OrganismQualifier> qualifier =
                 feature.getQualifiers("organism").stream().findFirst().map(q -> (OrganismQualifier) q);
 
-        return new GFF3Directives.GFF3Species(buildTaxonomyUrl(qualifier));
+        return new GFF3Species(buildTaxonomyUrl(qualifier));
     }
 
-    public GFF3Directives.GFF3SequenceRegion extractSequenceRegion(Entry entry) throws NoSourcePresentException {
+    public GFF3SequenceRegion extractSequenceRegion(Entry entry) throws NoSourcePresentException {
 
         String accession = entry.getPrimaryAccession();
```

Comment:
Since we are dealing with individual entries,  I think we can always rely on `entry.getSequence().getAccession()` instead of `entry.getPrimaryAccession()`

`String accession =  entry.getSequence().getAccession();`

We can throw an error if the accession is null.